lib/widgets/components/purchase/purchase_process.dart

```diff
@@ -0,0 +1,116 @@
+import 'package:coffeecard/base/strings.dart';
+import 'package:coffeecard/base/style/colors.dart';
+import 'package:coffeecard/cubits/purchase/purchase_cubit.dart';
+import 'package:coffeecard/errors/match_case_incomplete_exception.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class PurchaseProcess extends StatefulWidget {
+  @override
+  State<StatefulWidget> createState() => _PurchaseProcessState();
+}
+
+class _PurchaseProcessState extends State<PurchaseProcess>
+    with WidgetsBindingObserver {
+  AppLifecycleState? _notification;
+
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    setState(() {
+      _notification = state;
+    });
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance!.addObserver(this);
+  }
+
+  @override
+  void dispose() {
+    WidgetsBinding.instance!.removeObserver(this);
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Center(
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: [
+          BlocBuilder<PurchaseCubit, PurchaseState>(
+            builder: (context, state) {
+              final cubit = context.read<PurchaseCubit>();
+              if (_notification == AppLifecycleState.resumed) {
+                cubit.verifyPurchase();
+              }
+              if (state is PurchaseInitial) {
+                //Not related to previous check, hence a separate if statement
+                cubit.payWithMobilePay();
+                return makeDialog(Strings.purchaseTalking);
+              } else if (state is PurchaseProcessing ||
+                  state is PurchaseStarted) {
+                return makeDialog(Strings.purchaseTalking);
+              } else if (state is PurchaseVerifying) {
+                return makeDialog(Strings.purchaseCompleting);
+              } else if (state is PurchaseCompleted) {
+                return makeDialog(Strings.purchaseSuccess);
+              } else if (state is PurchasePaymentRejected) {
+                return makeDialog(
```

Comment:
How do we add the 'tap anywhere to dismiss' text to the rejected popup? seems like we may be able to reuse the same widget that the receipt card uses somehow?

Reply:
It was actually on purpose that it had not been added, since the user can't press anywhere to dismiss it. They can only dismiss it by pressing the "Ok". The reasoning being that we want them to acknowledge that an error happened, and not accidentally dismiss the dialogue